server/src/tpmctx.rs

```diff
@@ -5,13 +5,13 @@ use tpm2_rs_base::constants::TpmCc;
 use tpm2_rs_base::errors::TpmRcError;
 
 /// The object that processes incoming TPM requests and produces the corresponding TPM response.
-pub struct TpmContext<'a, Deps: TpmContextDeps> {
-    crypto: &'a mut Deps::Crypto,
+pub struct TpmContext<Deps: TpmContextDeps> {
```

Comment:
Optional: As the code is currently written, you don't technically need the constraint here on `Deps`, since you will anyways add the constraint in the `impl` block below.

It's only necessary to have the type constraint on the struct when someone can access the data directly (`pub` or has access to private data via being same module or submodule) from the struct. Otherwise, one can construct a `TpmContext` with an invalid type, but they can't actually use it they will get an error message (e.g. the error message changes slightly).

Optional, I don't mind much, just pointing it out.

Reply:
But that is the case here,

Command handler  is defined as:

```rust
pub struct CommandHandler<Deps: TpmContextDeps> {
    /// Gives access to cryptographic operations.
    pub crypto: Deps::Crypto,
}
```

and `TpmContext` is a wrapper around `CommandHandler` so it must have `Deps: TpmContextDeps`. Am I missing something?

Reply:
So I tried a bit, and I don't think that the type can be restricted, I am merging for the time being.